src/modals/UpdateReserve/index.tsx

```diff
@@ -0,0 +1,160 @@
+// Copyright 2023 @paritytech/polkadot-staking-dashboard authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+  ActionItem,
+  ButtonHelp,
+  ButtonPrimaryInvert,
+  ModalPadding,
+  ModalWarnings,
+} from '@polkadotcloud/core-ui';
+import { planckToUnit, unitToPlanck } from '@polkadotcloud/utils';
+import BigNumber from 'bignumber.js';
+import { useApi } from 'contexts/Api';
+import { useBalances } from 'contexts/Balances';
+import { useConnect } from 'contexts/Connect';
+import { useHelp } from 'contexts/Help';
+import { useModal } from 'contexts/Modal';
+import { useTransferOptions } from 'contexts/TransferOptions';
+import { Warning } from 'library/Form/Warning';
+import { Close } from 'library/Modal/Close';
+import { WithSliderWrapper } from 'modals/ManagePool/Wrappers';
+import Slider from 'rc-slider';
+import 'rc-slider/assets/index.css';
+import React, { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+
+export const UpdateReserve = () => {
+  const { t } = useTranslation('modals');
+  const { reserve, setReserve, getTransferOptions } = useTransferOptions();
+  const { units } = useApi().network;
+  const { activeAccount, accountHasSigner } = useConnect();
+  const { getLocks, getBalance } = useBalances();
+  const balance = getBalance(activeAccount);
+  const { network } = useApi();
+  const { setStatus, setResize } = useModal();
+  const allTransferOptions = getTransferOptions(activeAccount);
+  const { frozen } = balance;
+  const { openHelp } = useHelp();
+
+  // check account non-staking locks
+  const locks = getLocks(activeAccount);
+  const locksStaking = locks.find(({ id }) => id === 'staking');
+  const lockStakingAmount = locksStaking
+    ? locksStaking.amount
+    : new BigNumber(0);
+  // available balance data
+  const fundsFree = planckToUnit(
+    allTransferOptions.freeBalance.minus(frozen.minus(lockStakingAmount)),
+    units
+  );
+
+  const warnings = [];
+  if (fundsFree.isZero()) {
+    warnings.push(<Warning text={t('reserveLimit')} />);
```

Comment:
This text makes little sense and has no context - not understandable. If you are struggling to write English text reach out to myself to add text content.

Reply:
Please help to write the context in English if it is possible